host/xtest/regression_1000.c

```diff
@@ -1879,3 +1879,74 @@ static void xtest_tee_test_1023(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1023, xtest_tee_test_1023,
 		"Test ELF initialization (.init_array)");
+
+static void xtest_tee_test_1024(ADBG_Case_t *c)
+{
+	TEEC_Session session = { 0 };
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+	uint8_t *empty_buf = NULL;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &os_test_ta_uuid, NULL,
+			&ret_orig)))
+		return;
+
+	empty_buf = malloc(0);
+	if (!empty_buf) {
```

Comment:
From `malloc()` man page:
> If size is  0,  then malloc()  returns either NULL, or a unique pointer value that can later be successfully passed to free().

So i think here should better allocate a non-zero sized buffer to be sure to get a valid pointer.

Reply:
ok

Reply:
no sorry, you confuse me. This is the goal to have a buffer of 0 bytes allocated

Reply:
I don't agree. The goal here is to have a valid pointer (that is meaningful in the CA and that TEE driver can convert into a valid physical memory address) and to invoke a TA providing this pointer together with a size info set to 0.

So here, whatever memory is allocated, we don't care. We just care having a non-NULL pointer.
But `malloc()` man page says that `malloc()` may return NULL, depending of the libc, OS, ... Thus to ensure this test is portable, you must make sure `empty_buf` is not NULL, hence either malloc some byte(s), or use some valid reference, i.e.

```c
	char dummy_buf;

	empty_buf = &dummy_buf;
```